main.tf

```
@@ -1,65 +1,106 @@
-  ipam_scope_id                     = var.pool.ipam_scope_id
-  locale                            = var.pool.locale
-  publicly_advertisable             = var.pool.publicly_advertisable
-  public_ip_source                  = var.pool.public_ip_source
-  source_ipam_pool_id               = var.pool.source_ipam_pool_id
-  tags                              = var.pool.tags
+  description = var.vpc_ipam.ipam.description
+  dynamic "operating_regions" {
+    for_each = var.vpc_ipam.ipam.operating_regions
+    content {
+      region_name = operating_regions.value.region_name
+    }
+  }
+  tags    = var.vpc_ipam.ipam.tags
+  cascade = var.vpc_ipam.ipam.cascade
 }
 
-module "vpc_ipam_pool_cidr" {
-  source                               = "./modules/pool-cidr"
-  cidr                                 = var.pool_cidr.cidr
-  cidr_authorization_context_message   = var.pool_cidr.cidr_authorization_context_message
-  cidr_authorization_context_signature = var.pool_cidr.cidr_authorization_context_signature
-  ipam_pool_id                         = var.pool_cidr.ipam_pool_id
-  netmask_length                       = var.pool_cidr.netmask_length
-}
+resource "aws_vpc_ipam_pool" "this" {
+  for_each = { for pool in var.vpc_ipam.pools : pool.name => pool }
+
+  address_family                    = try(lower(each.value.address_family), null)
+  allocation_default_netmask_length = each.value.allocation_default_netmask_length
+  allocation_max_netmask_length     = each.value.allocation_max_netmask_length
+  allocation_min_netmask_length     = each.value.allocation_min_netmask_length
+  allocation_resource_tags          = each.value.allocation_resource_tags
+  auto_import                       = each.value.auto_import
+  aws_service                       = try(lower(each.value.aws_service), null)
+  description                       = each.value.description
+  ipam_scope_id = coalesce(
+    each.value.ipam_scope_name == "private_default_scope" ? aws_vpc_ipam.this[0].private_default_scope_id : null,
+    each.value.ipam_scope_name == "public_default_scope" ? aws_vpc_ipam.this[0].public_default_scope_id : null,
+    try(aws_vpc_ipam_scope.this[each.value.ipam_scope_name].id, null),
+    each.value.ipam_scope_id,
+  )
+  locale                = try(lower(each.value.locale), null)
+  publicly_advertisable = each.value.publicly_advertisable
+  public_ip_source      = each.value.public_ip_source
+  source_ipam_pool_id   = each.value.source_ipam_pool_id
+  tags                  = each.value.tags
 
-module "vpc_ipam_pool_cidr_allocation" {
-  source           = "./modules/pool-cidr-allocation"
-  cidr             = var.pool_cidr_allocation.cidr
-  disallowed_cidrs = var.pool_cidr_allocation.disallowed_cidrs
-  ipam_pool_id     = var.pool_cidr_allocation.ipam_pool_id
-  netmask_length   = var.pool_cidr_allocation.netmask_length
+  lifecycle {
+    precondition {
+      condition     = var.vpc_ipam.pools[0].ipam_scope_name != "" && var.vpc_ipam.pools[0].ipam_scope_id != ""
```

I _think_ we should have access to the for_each elements, so we can use `each.value` to test every item instead of just the `0` element...

```suggestion
      condition     = each.value.ipam_scope_name != null && each.value.ipam_scope_id != null
```